### DISCUSSION TITLE
Using api/system_status

### DISCUSSION THREAD

**Discussion by damclarkd** [jasonacox/Powerwall-Dashboard#43] [Ideas]
Hello Jason,
As I stated in a separate post, I am looking to consolidate a couple of running Tesla Powerwall Dashboard solutions into one instance. In the post I mentioned I have started a new install of your Powerwall Dashboard and moved my historical data into that instance. However I am not finding some of my data in your current configuration and would appreciate any suggestions on how to proceed.
The data in question all comes from the api/system_status call. Specifically - nominal_full_pack_energy and energy_charged, energy_discharged. energy_charged and energy_discharged are both lifetime values (I believe).
Looking at pypowerwall, it looks like api/system_status would be available. 
So would I amend the telegraf.conf to include this call?
Then add my retention policy batts_lt and cq for the test system data in influxdb.sql?

I do realize these changes move me out of sync with your code and I would not be able to use your upgrade.sh.

Does this sound like the way to go or am I missing something?
Any thoughts are appreciated.

Cheers!

**Comment by jasonacox** [comment]
Hi @damclarkd  - I think you have a great point here.  My original thought was to pull the `/pod` data to break the battery blocks down into separate trend lines for each Powerwall, but if you have configuration that it doesn't cover, I want to adjust.  Would you be willing to share the JSON output from the `http://localhost:8675/api/system_status` call (remove PackageSerialNumber)?  That will help me know how we are missing it on the `http://localhost:8675/pod` call.

Regardless of the above, there it may still make sense to add /api/system_status to InfluxDB.  I would be happy to add that to telegraf and influxdb.sql unless you want to do that and submit a PR. ;)

**Reply by damclarkd** [reply]
api/system_status results

{"command_source":"Configuration","battery_target_power":-5493.5,"battery_target_reactive_power":0,"nominal_full_pack_energy":27400,"nominal_energy_remaining":17458,"max_power_energy_remaining":0,"max_power_energy_to_be_charged":0,"max_charge_power":10000,"max_discharge_power":10000,"max_apparent_power":10000,"instantaneous_max_discharge_power":14000,"instantaneous_max_charge_power":14000,"instantaneous_max_apparent_power":20460,"grid_services_power":-0,"system_island_state":"SystemGridConnected","available_blocks":2,"battery_blocks":[{"Type":"","PackagePartNumber":"1092170-05-J","PackageSerialNumber":,"disabled_reasons":[],"pinv_state":"PINV_GridFollowing","pinv_grid_state":"Grid_Compliant","nominal_energy_remaining":8737,"nominal_full_pack_energy":13752,"p_out":-2780.0000000000005,"q_out":20,"v_out":244.8,"f_out":59.986000000000004,"i_out":58.300000000000004,"energy_charged":2967610,"energy_discharged":2535170,"off_grid":false,"vf_mode":false,"wobble_detected":false,"charge_power_clamped":false,"backup_ready":true,"OpSeqState":"Active","version":"a54352e6706386"},{"Type":"","PackagePartNumber":"1092170-05-J","PackageSerialNumber":"","disabled_reasons":[],"pinv_state":"PINV_GridFollowing","pinv_grid_state":"Grid_Compliant","nominal_energy_remaining":8721,"nominal_full_pack_energy":13648,"p_out":-2740,"q_out":20,"v_out":244.9,"f_out":59.985,"i_out":57,"energy_charged":2898090,"energy_discharged":2467520,"off_grid":false,"vf_mode":false,"wobble_detected":false,"charge_power_clamped":false,"backup_ready":true,"OpSeqState":"Active","version":"a54352e6706386"}],"ffr_power_availability_high":11600,"ffr_power_availability_low":11600,"load_charge_constraint":0,"max_sustained_ramp_rate":2500000,"grid_faults":[],"can_reboot":"Power flow is too high","smart_inv_delta_p":0,"smart_inv_delta_q":0,"last_toggle_timestamp":"2022-06-22T14:15:00.085325336-07:00","solar_real_power_limit":-1,"score":10000,"blocks_controlled":2,"primary":true,"auxiliary_load":0,"all_enable_lines_high":true,"inverter_nominal_usable_power":11600,"expected_energy_remaining":0}

**Comment by damclarkd** [comment]
Hello Jason,
Regarding adding /api/system_status to InfluxDB, that is something I have done with success on my test system. However never having done a PR here, not sure how I would proceed. I can read up on the process, just not sure where to start exactly. Willing to learn.
Cheers
